lib/semian.rb

```diff
@@ -150,6 +151,7 @@ def to_s
   #
   # Returns the registered resource.
   def register(name, **options)
+    thread_safe = options[:thread_safety_disabled]
```

Comment:
I this legit? I feel like that might be a bit hacky

Reply:
For backward compatibility, you'll need to support both, unfortunately. However, you can emit to `STDERR` that this option per resource will be removed in favour of a global one.